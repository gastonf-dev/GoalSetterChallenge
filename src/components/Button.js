/**
 * @flow stric-local
 * @format
 */

import React from 'react';
import type {Node} from 'react';

import {TouchableOpacity, StyleSheet, Text, Platform} from 'react-native';
import COLORS from '../styles/Colors';

type Props = {
  title: string,
  onPress: () => mixed,
  disabled?: boolean,
  large?: boolean,
};

const Button = ({title, onPress, disabled, large}: Props): Node => {
  return (
    <>
      <TouchableOpacity
        style={[
          styles.button,
          disabled ? {backgroundColor: COLORS.gray} : null,
          large ? {width: '90%'} : null,
        ]}
        onPress={onPress}
        disabled={disabled}
        testID="button">
        <Text style={styles.buttonText}>{title}</Text>
      </TouchableOpacity>
    </>
  );
};

export default Button;

const styles = StyleSheet.create({
  button: {
    backgroundColor: COLORS.green,
    width: '60%',
    alignSelf: 'center',
    marginTop: 24,
  },
  buttonText: {
    fontFamily: Platform.OS === 'ios' ? 'Avenir LT Std' : 'AvenirLTStd',
    fontSize: 12,
    textAlign: 'center',
    color: 'white',
    paddingVertical: 12,
  },
});
