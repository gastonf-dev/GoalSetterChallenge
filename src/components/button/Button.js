/**
 * @flow stric-local
 * @format
 */

import React from 'react';
import type {Node} from 'react';

import {
  Button as ButtonRN,
  TouchableOpacity,
  StyleSheet,
  Text,
} from 'react-native';
import COLORS from '../../styles/Colors';

type Props = {
  title: string,
  onPress: function,
  disabled?: boolean,
  large?: boolean,
};

const Button = ({title, onPress, disabled, large}: Props): Node => {
  return (
    <>
      <TouchableOpacity
        style={[
          styles.button,
          disabled ? {backgroundColor: COLORS.gray} : null,
          large ? {width: '90%'} : null,
        ]}
        onPress={onPress}
        disabled={disabled}
        testID="button">
        <Text style={styles.buttonText}>{title}</Text>
      </TouchableOpacity>
    </>
  );
};

export default Button;

const styles = StyleSheet.create({
  button: {
    backgroundColor: COLORS.green,
    width: '60%',
    alignSelf: 'center',
    marginTop: 24,
  },
  buttonText: {
    fontFamily: 'AvenirLTStd',
    fontSize: 12,
    textAlign: 'center',
    color: 'white',
    paddingVertical: 12,
  },
});
