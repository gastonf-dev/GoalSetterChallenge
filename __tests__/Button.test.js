import React from 'react';
import renderer from 'react-test-renderer';

import {render, fireEvent, waitFor} from '@testing-library/react-native';

import Button from '../src/components/button/Button';

describe('<Button />', () => {
  test('render button enable and short', () => {
    const ENABLE_SHORT = {
      title: 'Test button',
      onPress: () => console.log('onPress'),
      disabled: false,
      large: false,
    };
    const wrapper = renderer.create(<Button {...ENABLE_SHORT} />);
    expect(wrapper.getInstance);
  });

  test('render button disable and short', () => {
    const DISABLE_SHORT = {
      title: 'Test button',
      onPress: () => console.log('onPress'),
      disabled: true,
      large: false,
    };
    const wrapper = renderer.create(<Button {...DISABLE_SHORT} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('render button enable and large', () => {
    const ENABLE_LARGE = {
      title: 'Test button',
      onPress: () => console.log('onPress'),
      disabled: false,
      large: true,
    };

    const wrapper = renderer.create(<Button {...ENABLE_LARGE} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('render button disable and large', () => {
    const DISABLE_LARGE = {
      title: 'Test button',
      onPress: () => console.log('onPress'),
      disabled: true,
      large: true,
    };

    const wrapper = renderer.create(<Button {...DISABLE_LARGE} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('render button enable and short and press it', async () => {
    const ON_PRESS_BUTTON = jest.fn();
    const ENABLE_LARGE = {
      title: 'Test button',
      onPress: ON_PRESS_BUTTON,
      disabled: false,
      large: true,
    };

    const {getByTestId} = render(<Button {...ENABLE_LARGE} />);
    fireEvent.press(getByTestId('button'));
    await waitFor(() => {
      expect(ON_PRESS_BUTTON).toHaveBeenCalled();
    });
  });
});
