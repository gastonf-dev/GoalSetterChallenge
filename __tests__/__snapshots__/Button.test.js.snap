// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> render button disable and large 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#D8D8D8",
      "marginTop": 24,
      "opacity": 1,
      "width": "90%",
    }
  }
  testID="button"
>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "AvenirLTStd",
        "fontSize": 12,
        "paddingVertical": 12,
        "textAlign": "center",
      }
    }
  >
    Test button
  </Text>
</View>
`;

exports[`<Button /> render button disable and short 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#D8D8D8",
      "marginTop": 24,
      "opacity": 1,
      "width": "60%",
    }
  }
  testID="button"
>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "AvenirLTStd",
        "fontSize": 12,
        "paddingVertical": 12,
        "textAlign": "center",
      }
    }
  >
    Test button
  </Text>
</View>
`;

exports[`<Button /> render button enable and large 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#5FBA20",
      "marginTop": 24,
      "opacity": 1,
      "width": "90%",
    }
  }
  testID="button"
>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "AvenirLTStd",
        "fontSize": 12,
        "paddingVertical": 12,
        "textAlign": "center",
      }
    }
  >
    Test button
  </Text>
</View>
`;
